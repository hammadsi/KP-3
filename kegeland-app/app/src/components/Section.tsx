import React from 'react';
import {StyleSheet, ViewStyle} from 'react-native';
import {Divider, Surface, Title} from 'react-native-paper';

export type SectionProps = {
  title: string;
  style?: ViewStyle;
  children?: React.ReactNode;
};

/**
 * Component for rendering a section.
 * @param props the props
 * @see {@link SectionProps}
 */
const Section: React.FC<SectionProps> = ({title, children, ...props}) => {
  return (
    <Surface style={[styles.surface, props.style]}>
      <Title style={styles.title}>{title}</Title>
      <Divider />
      {children}
    </Surface>
  );
};

const styles = StyleSheet.create({
  surface: {
    margin: 10,
    paddingHorizontal: 10,
    borderRadius: 15,
  },
  title: {
    fontSize: 15,
    fontWeight: '700',
    paddingHorizontal: 10,
    paddingVertical: 5,
  },
});

export default Section;
