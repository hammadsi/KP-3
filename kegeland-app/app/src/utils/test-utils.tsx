/* istanbul ignore file */
import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {Provider as PaperProvider} from 'react-native-paper';
import {render, RenderOptions} from '@testing-library/react-native';
import {Provider as ReduxProvider} from 'react-redux';

import {store} from '~state/store';
import {lightTheme} from '~constants/theme';
import {initialStore, mockStore} from '~state/ducks/__mocks__/store.mock';

type TestWrapperProps = {
  children: React.ReactNode;
  store?: typeof store;
};

/**
 * Wrapper component with redux state. Used when testing components
 * which requires redux.
 * @param param0 test wrapper props @see {@link TestWrapperProps}
 * @returns
 */
export const TestWrapper: React.FC<TestWrapperProps> = ({
  children,
  ...props
}) => {
  return (
    <ReduxProvider store={props.store || mockStore(initialStore)}>
      <PaperProvider theme={lightTheme}>
        <NavigationContainer theme={lightTheme}>{children}</NavigationContainer>
      </PaperProvider>
    </ReduxProvider>
  );
};

/**
 * Custom test renderer which wraps components with a redux store provider
 * @param ui the component to render
 * @param options renderer options
 * @returns test rendered componnet
 */
export const customRender = <T extends any>(
  ui: React.ReactElement<T>,
  options?: {
    wrapperProps?: {
      store: any;
    };
    renderOptions?: RenderOptions;
  },
) =>
  render(ui, {
    wrapper: (props) => <TestWrapper {...props} {...options?.wrapperProps} />,
    ...options?.renderOptions,
  });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
