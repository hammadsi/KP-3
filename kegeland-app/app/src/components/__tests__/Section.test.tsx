import React from 'react';

import {render} from '~utils/test-utils';

import Section, {SectionProps} from '../Section';

describe('Test Section-component', () => {
  it('should render correctly', () => {
    const props: SectionProps = {
      title: 'Test section',
    };
    const component = <Section {...props} />;
    const tree = render(component);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
