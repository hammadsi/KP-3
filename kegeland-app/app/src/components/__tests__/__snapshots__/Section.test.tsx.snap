// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Section-component should render correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 15,
        "margin": 10,
        "paddingHorizontal": 10,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 4,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              Object {
                "fontSize": 15,
                "fontWeight": "700",
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              },
            ],
          ],
        ]
      }
    >
      Test section
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.12)",
            "height": 0.5,
          },
          undefined,
          undefined,
        ]
      }
    />
  </View>
</View>
`;
